jetty-client/src/test/java/org/eclipse/jetty/client/IncorrectRedirectServer.java

```diff
@@ -0,0 +1,79 @@
+package org.eclipse.jetty.client;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class IncorrectRedirectServer
+{
+    private final Thread serverThread;
+    private final AtomicInteger port;
+
+    public IncorrectRedirectServer(final String redirect)
+    {
+        this.port = new AtomicInteger(0);
+        this.serverThread = new Thread(new Runnable()
+        {
+            @Override
+            public void run()
+            {
+                try (ServerSocket serverSocket = new ServerSocket())
+                {
+                    serverSocket.bind(new InetSocketAddress(InetAddress.getLocalHost(), 0));
+                    port.set(serverSocket.getLocalPort());
+
+                    while (!Thread.currentThread().isInterrupted())
+                    {
+                        try (Socket clientSocket = serverSocket.accept();
+                             PrintWriter writer = new PrintWriter(clientSocket.getOutputStream(), false);
```

Comment:
Move the creation of the `PrintWriter` where it is used, i.e. after you have read the request.